### PR TITLE
Small improvements

### PR DESCRIPTION
**Pull request by pruchay**
- Fixed problem with links on pages; 
- Used for...of instead of for statement; 
- Fixed problem with CSS classes then the active class is given; 
- Add .idea folder to .gitignore

### CODE REVIEW

File: js/route-now.js [hunk 1]
```diff
@@ -28,9 +28,10 @@ var $Router = {},
         }
     },
     resetClassForAnchors = () => {
-        var listOfAnchorElems = document.querySelectorAll("a." + $Router.options.activateLinkClass);
-        for (var i = 0; i < listOfAnchorElems.length; i++) {
-            listOfAnchorElems[i].setAttribute("class", $Router.options.defaultLinkClass);
+        let listOfAnchorElems = document.querySelectorAll("a." + $Router.options.activateLinkClass);
+        for (let element of listOfAnchorElems) {
+            let classes = element.className.replace(/active/, "").trim();
+            element.setAttribute("class", classes + " " + $Router.options.defaultLinkClass);
```
Comment (pruchay) [inline js/route-now.js:34]: I think this way it will be more clearer :)

File: js/route-now.js [hunk 2]
```diff
@@ -90,9 +91,12 @@ $Router.options = {
...
     hashPath = hashPath.replace(/#/g, '');
-    $Router.options.activateLinks && document
-        .querySelector("a[r-href='" + hashPath + "']")
-        .setAttribute("class", $Router.options.defaultLinkClass + " " + $Router.options.activateLinkClass);
+
+    let linkQuery = document.querySelector("a[r-href='" + hashPath + "']");
+
+    $Router.options.activateLinks
+        && linkQuery
+        && linkQuery.setAttribute("class", linkQuery.className + " " + $Router.options.defaultLinkClass + " " + $Router.options.activateLinkClass);
```
Comment (pruchay) [inline js/route-now.js:99]: I added this because without this if the link has a CSS class it will be overridden.

File: js/script.js [hunk 3]
```diff
@@ -24,6 +24,12 @@ $Router.config([
                 })
             });
         }
+
+        let anchorList = Array.from(document.querySelectorAll('a[r-href]'));
+        anchorList.forEach(anchor => {
+            let url = anchor.getAttribute('r-href');
+            anchor.setAttribute('href', '#' + url);
+        });
```
Comment (pruchay) [inline js/script.js:32]: Without this forEach link's on pages doesn't work. 
You can check it, for example, add link `<a r-href="home">Installation</a>` at `partial/options.html`.

**Comment by pruchay** [comment]
Also, don't forget to minify the `route-now.js` file ;) Or if you want, I can do it by myself.

**Comment by akhilarjun** [comment]
Hey, thanks a lot for the PR. Sorry, I just saw this now. Can you also minify the route-now library and push it out?

**Comment by pruchay** [comment]
> Hey, thanks a lot for the PR. Sorry, I just saw this now. Can you also minify the route-now library and push it out?

I minified `route-now.js` file.

**Pull request merged**